agent/banner/banner.go

```diff
@@ -174,14 +174,14 @@ func (w *bannerResponseWriter) WriteHeader(statusCode int) {
 		return
 	}
 	w.wroteHeader = true
-	if !isFrameableHTMLResponse(statusCode, w.Header()) {
-		w.wrapped.WriteHeader(statusCode)
-		w.writeBytes = true
-		return
-	}
-	setNotCacheable(w.Header())
-	setXFrameOptionsSameOrigin(w.Header())
-	if w.isAlreadyFramed {
+	// Set X-Frame-Options header to sameorigin if the page is already framed
+	// or if the page is frameable.
+	if w.isAlreadyFramed || isFrameableHTMLResponse(statusCode, w.Header()) {
+		setNotCacheable(w.Header())
```

Comment:
Jerry - what is the reasoning behind setting the not cacheable header, and do the same conditions apply?

Reply:
Discussed offline - the point is to not cache the wrapper html - so we should only set this if we're gonna wrap the page - will fix